message-broker/src/lib.rs

```diff
@@ -116,6 +124,50 @@ decl_module! {
 				);
 				*up = up.split_off(num);
 			});
+
+			// Sending HRMP messages is a little bit more involved. On top of the number of messages
+			// per block limit, there is also a constraint that it's possible to send only a single
+			// message to a given recipient per candidate.
+			let mut non_empty_hrmp_channels = <Self as Store>::NonEmptyHrmpChannels::get();
```

Comment:
maybe a dumb question, but could all of this logic happen at the beginning of the next block?